src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -34,6 +38,8 @@
     private final ObservableList<BibEntry> selectedEntries = FXCollections.observableArrayList();
     private final ObservableMap<BibDatabaseContext, ObservableList<GroupTreeNode>> selectedGroups = FXCollections.observableHashMap();
     private final OptionalObjectProperty<SearchQuery> activeSearchQuery = OptionalObjectProperty.empty();
+    private final IntegerProperty searchResultSize = new SimpleIntegerProperty();
+    private final SearchQueryHighlightObservable searchQueryHighlightObservable = new SearchQueryHighlightObservable();
```

Comment:
I guess `searchQuery` is slightly better variable name.

Reply:
Moreover, why not simply reuse `activeSearchQuery` instead of using a custom class for the same purpose?
(Move `getPatternForWords` to a new `HlighlightHelper` class or something).

Reply:
Yeah, I tried around with that and in general you are right. However, the searchQuery property is of course only triggered once, but when I change the tab or the entry the searchQuery needs to be retriggered or at least the pattern must be returned again.

The solution to solve this without redoing the searchQuery is the EventBus listener construct.
Probably this is the reason why the class exists. the fireSearchQuery gets the pattern and updates all listeners.
```
    public void fireSearchlistenerEvent(SearchQuery searchQuery) {
        Objects.requireNonNull(searchQuery);

        // Parse the search string to words
        pattern = getPatternForWords(searchQuery.getSearchWords(), searchQuery.isRegularExpression(),
                searchQuery.isCaseSensitive());

        update(); //calls eventBus.post
    }

```

Reply:
> but when I change the tab or the entry the searchQuery needs to be retriggered or at least the pattern must be returned again.

I don't really understand why this is. The search query didn't change, and thus it is not the job of the statemanager to fire additional events. The components (e.g. the entry editor) is responsible to update its state accordingly (everything else would be a wrong assignment of responsibilities).

Reply:
Got it. Put it in the SearchQuery class. SourceTab was easy going. Preview needed a bit more thinking, but works now.